### PR TITLE
crimson/os/alienstore: use homebrew counting_semaphore

### PR DESCRIPTION
**Pull request by xxhdx1985126**
std::counting_semaphore has the bug: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=104928, go back to our homebrew version

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by athanatos — APPROVED** [review]
Nit: can we add a comment explaining why we're defaulting to the homebrew one here?  Preferably with a link to the gcc issue tracker.

File: src/crimson/os/alienstore/thread_pool.h [hunk 1]
```diff
@@ -15,14 +15,7 @@
 #include <seastar/core/semaphore.hh>
 #include <seastar/core/sharded.hh>
 
-#if __cplusplus > 201703L
-#include <semaphore>
-namespace crimson {
-  using std::counting_semaphore;
-}
-#else
```
Comment (tchaikov) [inline src/crimson/os/alienstore/thread_pool.h:23]: i concur with Sam, better off adding a comment explaining the rationales behind this.

Comment (xxhdx1985126) [reply src/crimson/os/alienstore/thread_pool.h:23]: Yeah, will do.

**Comment by xxhdx1985126** [comment]
@tchaikov @athanatos Done, please take a look again:-) Thanks.

**Review by athanatos — APPROVED** [review]

**Review by liu-chunmei — APPROVED** [review]

**Review by tchaikov — APPROVED** [review]
lgtm

**Comment by Matan-B** [comment]
https://pulpito.ceph.com/matan-2024-03-05_16:18:08-crimson-rados-wip-matanb-crimson-testing-march-5-distro-crimson-smithi/

`7582495` - osd.1 - https://tracker.ceph.com/issues/64009
`7582487` - osd.0 - https://tracker.ceph.com/issues/64009

**Pull request merged**